internal/metrics/metrics.go

```diff
@@ -0,0 +1,290 @@
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_creation_total",
+			Help: "Total number of EvictionAutoScaler resources created",
+		},
+		[]string{"namespace", "pdb_name", "target_deployment"},
+	)
+
+	// NodeCordoningCounter tracks node cordoning events detected
+	// Labels: cordoned (true/false)
+	NodeCordoningCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_node_cordoning_total",
+			Help: "Total number of node cordoning events detected by the eviction autoscaler",
+		},
+		[]string{"cordoned"},
+	)
+
+	// OldNotReadyPodsGauge tracks pods that are old and not ready
+	// Labels: namespace, target_name, likely_to_help (true/false)
+	OldNotReadyPodsGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eviction_autoscaler_old_not_ready_pods",
+			Help: "Number of old pods that are not ready",
+		},
+		[]string{"namespace", "target_name", "likely_to_help"},
+	)
+
+	// MaxUnavailableGauge tracks the max unavailable value from PDBs
+	// Labels: namespace, pdb_name, target_name
+	MaxUnavailableGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eviction_autoscaler_max_unavailable",
+			Help: "Max unavailable value from PDB configuration",
+		},
+		[]string{"namespace", "pdb_name", "target_name"},
+	)
+
+	// MinAvailableEqualsReplicasGauge tracks when minAvailable equals replicas (indicating PDB might be too strict)
+	// Labels: namespace, pdb_name, target_name
+	MinAvailableEqualsReplicasGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eviction_autoscaler_min_available_equals_replicas",
+			Help: "Indicates when minAvailable equals replicas (1 if true, 0 if false)",
+		},
+		[]string{"namespace", "pdb_name", "target_name"},
+	)
+
+	// PDBCounter tracks the number of PDBs with an increment interface
+	// Labels: namespace, created_by_us (true/false)
+	PDBCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_pdb_count_total",
+			Help: "Total count of PDBs processed by the eviction autoscaler",
+		},
+		[]string{"namespace", "created_by_us"},
+	)
+)
+
+// Constants for PDB creation tracking
+const (
+	PDBCreatedByUs    = true
+	PDBNotCreatedByUs = false
+)
+
+// Constants for deployment tracking
+const (
+	CanCreatePDB    = true
+	CannotCreatePDB = false
+)
+
+func init() {
+	// Register metrics with controller-runtime's registry
+	metrics.Registry.MustRegister(
+		DeploymentGauge,
+		PDBGauge,
+		EvictionCounter,
+		BlockedEvictionCounter,
+		ScalingOpportunityCounter,
+		ActualScalingCounter,
+		PDBCreationCounter,
+		EvictionAutoScalerCreationCounter,
+		NodeCordoningCounter,
+		OldNotReadyPodsGauge,
+		MaxUnavailableGauge,
+		MinAvailableEqualsReplicasGauge,
+		PDBCounter,
+	)
+}
+
+// Helper functions to update metrics
+
+// IncrementDeploymentCount increments the deployment gauge
+func IncrementDeploymentCount(namespace string, canCreatePDB bool) {
```

Comment:
instead of exposing two funcitons for each incrment/decrment can just expose DeploymentGauge.WithLabelValues(namespace, canCreatePDBStr)

Or since the counters are already public are you getting much value should you hid them?